### PR TITLE
Feature/tax exporter

### PR DESCRIPTION
**Pull request by fenris85**

### CODE REVIEW

File: src/balances/balances.ts [hunk 1]
```diff
@@ -199,6 +200,12 @@ export function createCombinedBalances$(
   etherPriceUsd$: Observable<BigNumber>,
   transactions$: Observable<TxState[]>
 ): Observable<CombinedBalances> {
+
```
Comment (maciejka) [inline src/balances/balances.ts]: Should be removed.

File: src/balances/taxExporter.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { BigNumber } from 'bignumber.js';
+import { curry } from 'lodash';
+import { combineLatest, Observable } from 'rxjs';
+import { flatMap, switchMap } from 'rxjs/operators';
+import { proxyAddress$ } from '../blockchain/calls/instant';
+import { NetworkConfig } from '../blockchain/config';
+import { vulcan0x } from '../blockchain/vulcan0x';
+import { web3 } from '../blockchain/web3';
+
+export interface TaxExportItem {
```
Comment (maciejka) [inline src/balances/taxExporter.ts]: What TaxExportItem and TaxExport is for?

Comment (fenris85) [reply src/balances/taxExporter.ts]: You are right. It's not necessary - not in use anymore. I will remove it.

File: src/balances/balances.ts (hunk 1 above)
Comment (fenris85) [reply src/balances/balances.ts]: Tslint says that:
> file should end with a newline

File: src/blockchain/calls/instant.tsx [hunk 3]
```diff
@@ -285,10 +286,17 @@ export const offers: CallDef<OffersData, any> = {
 const nullAddress = '0x0000000000000000000000000000000000000000';
 export function proxyAddress$(
   context: NetworkConfig,
-  account: string
+  account: string,
+  proxyRegistryAddress?: string
 ): Observable<string | undefined> {
-  return bindNodeCallback(context.instantProxyRegistry.contract.proxies)(account).pipe(
+  return bindNodeCallback(
+    (proxyRegistryAddress ?
```
Comment (peculiarity) [inline src/blockchain/calls/instant.tsx]:
```javascript
(
 proxyRegistryAddress
    ? web3.eth.contract(proxyRegistry as any).at(proxyRegistryAddress)
    :  context.instantProxyRegistry.contract
).proxies
```

File: src/blockchain/config.ts [hunk 4]
```diff
@@ -145,6 +146,8 @@ const kovan: NetworkConfig = {
   etherscan: {
     url: 'https://kovan.etherscan.io',
   },
+  // taxProxyRegistries: ['0x383a7fc29edde64aec7f776e2517ec8819e147f1']
```
Comment (peculiarity) [inline src/blockchain/config.ts]: Why do we have one commented value and one that is actually use? If you don't need it just remove it

**Comment by peculiarity** [comment]
When I try to export my trades I have Bad Request on the vulcan0x call.
Also I have `process.env.REACT_APP_GRAPHQL_DEVMODE undefined` under kovan

File: src/balances/TaxExporterButton.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    return (
+      <Button
+        onClick={this.exportTrades}
+      >
+        Export trades
+      </Button>);
+  }
+
+  private exportTrades = () => {
+    // const currentDate = new Date();
```
Comment (peculiarity) [inline src/balances/TaxExporterButton.tsx]: If you don't need them , remove them

File: src/blockchain/config.ts (hunk 4 above)
Comment (fenris85) [reply src/blockchain/config.ts]: Sure!

File: src/balances/TaxExporterButton.tsx (hunk 5 above)
Comment (fenris85) [reply src/balances/TaxExporterButton.tsx]: Removing it

File: src/blockchain/calls/instant.tsx (hunk 3 above)
Comment (fenris85) [reply src/blockchain/calls/instant.tsx]: What is your concern about it? You mean that for kovan **instantProxyRegistry** and **taxProxyRegistries** are the same? That's right, however for mainnet they will be different (there was different ProxyRegistry on June 2018).

**Comment by fenris85** [comment]
> When I try to export my trades I have Bad Request on the vulcan0x call.
> Also I have `process.env.REACT_APP_GRAPHQL_DEVMODE undefined` under kovan

Have you set up .env var? Shoud be `REACT_APP_GRAPHQL_DEVMODE=2232759874`

File: src/blockchain/calls/instant.tsx (hunk 3 above)
Comment (peculiarity) [reply src/blockchain/calls/instant.tsx]: My concern is the poorly formatted code :)

File: src/balances/balances.ts (hunk 1 above)
Comment (fenris85) [reply src/balances/balances.ts]: removed

File: src/blockchain/calls/instant.tsx (hunk 3 above)
Comment (fenris85) [reply src/blockchain/calls/instant.tsx]: got it. Fixed.

**Pull request merged**